z_infra_provisioning/cloud_infra_center/ocp_upi/wait-for-install-complete.yaml

```diff
@@ -0,0 +1,80 @@
+# =================================================================
+# Copyright 2021 https://github.com/openshift/installer
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =================================================================
+
+# Required Python packages:
+#
+# ansible
+# openstackclient
+# openstacksdk
+
+- hosts: localhost
+  gather_facts: no
+
+  tasks:
+  - name: oc path to configs.imageregistry.operator.openshift.io/cluster
+    command: ./oc patch configs.imageregistry.operator.openshift.io/cluster --type merge --patch '{"spec":{"storage":{"emptyDir":{}}}}'
+    environment:
+      KUBECONFIG: "./auth/kubeconfig"
+    retries: 100
+    delay: 10
+    register: result
+    until: result is not failed
+
+  - name: Approve the pending CSR
+    shell: |
+        export KUBECONFIG=./auth/kubeconfig
+        for i in `./oc get csr --no-headers | grep -i pending |  awk '{ print $1 }'`; do ./oc adm certificate approve $i; done
```

Comment:
@shhull When you try to approve CSR, you'll find the `Pending CSRs` may not appear at the same time. For example, you approve work-0's `Pending CSR`, but the work-1's `Pending CSR` may appear after this shell command, so the work-1's `Pending CSR` will be ignored.

Reply:
yes, you're right, so I've added another shell command to confirm all pending should be approved after all worker nodes are adding, we've another python code to wait node csrs in configure-compute-nodes task.
```- name: 'Approve worker nodes csr'
  script: tools/wait-nodes-csrs.py {{ os_compute_server_name }}-{{ item.1 }}
  args:
    executable: python3
  with_indexed_items: "{{ compute_nodes_json.compute_node_names }}"
```
are you suggesting use recycling to execute shell command to approve all nodes?

Reply:
@shhull Sorry, it works in this way. I think it's better to use tools/wait-nodes-csrs.py only in wait-for-install-complete.yaml, but it works in your way.